ir/shared/src/main/scala/org/scalajs/ir/Trees.scala

```diff
@@ -168,6 +168,33 @@ object Trees {
   sealed case class If(cond: Tree, thenp: Tree, elsep: Tree)(val tpe: Type)(
       implicit val pos: Position) extends Tree
 
+  sealed case class LinkTimeIf(val cond: LinkTimeCondition, val thenp: Tree,
+      val elsep: Tree)(val tpe: Type)(implicit val pos: Position) extends Tree
+
+  sealed abstract class LinkTimeCondition {
+    val pos: Position
+  }
+
+  object LinkTimeCondition {
+    final case class Binary(op: BinaryOp.Code, lhs: LinkTimeValue,
+        rhs: LinkTimeValue)(implicit val pos: Position) extends LinkTimeCondition
+  }
+
+  sealed abstract class LinkTimeValue {
+    val tpe: Type
+  }
+
+  object LinkTimeValue {
+    final case class Property(name: String, tpe: Type)
+      extends LinkTimeValue
+    final case class IntConst(v: Int) extends LinkTimeValue {
+      val tpe = IntType
+    }
+    final case class BooleanConst(v: Boolean) extends LinkTimeValue {
+      val tpe = BooleanType
+    }
+  }
```

Comment:
Does it make sense to attempt to re-use trees here? (and maybe have a marker trait?)

Reply:
You mean reusing `BooleanLiteral`s here?
I believe it's more appropriate to have separate trees for this use-case, even if the data these trees hold is same as Literal trees. (Previously, I reused `BinaryOp.Code` for `LinkTimeTree.BinaryOp`, but I've since stopped reusing it and defined a separate set of operation codes for link-time binary operations.)

The reason is that the "runtime behavior" of `Literal`s and these `(Boolean|Int)Constant`s are different. The latter appear only in `linkTimeIf` statements and are evaluated by the linker, not at runtime."

Ended up like this, and maybe `LinkTimeTree` works as a marker trait?

```scala
  sealed abstract class LinkTimeTree {
    val pos: Position
    val tpe: Type
  }

  object LinkTimeTree {
    final case class BinaryOp(op: LinkTimeOp.Code, lhs: LinkTimeTree,
        rhs: LinkTimeTree)(implicit val pos: Position) extends LinkTimeTree {
            val tpe = BooleanType
        }
    final case class Property(name: String, tpe: Type)(implicit val pos: Position)
      extends LinkTimeTree
    final case class IntConst(v: Int)(implicit val pos: Position) extends LinkTimeTree {
      val tpe = IntType
    }
    final case class BooleanConst(v: Boolean)(implicit val pos: Position) extends LinkTimeTree {
      val tpe = BooleanType
    }
  }
```